protoc-gen-swagger/genswagger/template.go

```diff
@@ -292,6 +312,92 @@ func renderMessagesAsDefinition(messages messageMap, d swaggerDefinitionsObject,
 	}
 }
 
+//AddStreamError Adds internal.StreamError and Any to registry for stream responses
+func AddStreamError(reg *descriptor.Registry) error {
+	//load internal protos
+	any, err := fileDescriptorProtoFromProtoDescriptor(&any.Any{})
+	if err != nil {
+		return err
+	}
+	streamError, err := fileDescriptorProtoFromProtoDescriptor(&internal.StreamError{})
+	if err != nil {
+		return err
+	}
+	if err := reg.Load(&plugin.CodeGeneratorRequest{
+		ProtoFile: []*pbdescriptor.FileDescriptorProto{
+			any,
+			streamError,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+type protoDescriptor interface {
+	Descriptor() ([]byte, []int)
+}
+
+func fileDescriptorProtoFromProtoDescriptor(pd protoDescriptor) (*pbdescriptor.FileDescriptorProto, error) {
+	pdd, _ := pd.Descriptor()
+	r, err := gzip.NewReader(bytes.NewReader(pdd))
+	if err != nil {
+		return nil, err
+	}
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(r)
+	if err != nil {
+		return nil, err
+	}
+	err = r.Close()
+	if err != nil {
+		return nil, err
+	}
+	fdp := &pbdescriptor.FileDescriptorProto{}
+	if err := proto.Unmarshal(buf.Bytes(), fdp); err != nil {
+		return nil, err
+	}
+	//hide the fact that we are loading this from the pb.go instead of the proto directly
+	fdp.SourceCodeInfo = &pbdescriptor.SourceCodeInfo{}
```

Comment:
The whole thing or just setting SourceCodeInfo? 😄

The purpose of the function in general is to load in the StreamError and Any messages into the proto registry from memory because we won't have any idea where the proto files are on the end user's system. Every compiled proto generates a Descriptor function which returns a gzipped byte slice of the FileDescriptorProto which is exactly what we need. Unfortunately to save space they drop the SourceCodeInfo before gzipping up the proto and writing the bytes to the pb.go which leads into the need for this line.

We need to set SourceCodeInfo unfortunately there is a check for SourceCodeInfo being non nil when the swagger generator tries to use the SourceCodeInfo in protoComments()
```golang
if file.SourceCodeInfo == nil {
    fmt.Fprintln(os.Stderr, "descriptor.File should not contain nil SourceCodeInfo")
    return ""
}
```
As long as it's not nil that err log wont print and the end user wont be any wiser that we are loading these in from the generated pb.go Descriptor() gzipped FileDescriptorProto. Unfortunately there wont be any descriptions that come with the StreamError or Any that we are loading in in this way not that it really matters for StreamError because there isn't many comments in that proto file. Bright side is if the end user loads in any.proto it should overwrite our any.pb.go and pull in the comments. I'm gonna go test that theory right now, be back with the results

Reply:
yep works perfectly

Reply:
Amazing, thanks!